packages/yoroi-extension/app/UI/features/swap-new/common/components/Modals/DisclaimerDialog.tsx

```diff
@@ -0,0 +1,135 @@
+import { useEffect, useState } from 'react';
+import { useStrings } from '../../hooks/useStrings';
+import { Stack, Typography, Checkbox, FormControlLabel, Box } from '@mui/material';
+import { useModal } from '../../../../../components/modals/ModalContext';
+import { LoadingButton } from '@mui/lab';
+import LocalStorageApi from '../../../../../../api/localStorage/index';
+import { useNavigateTo } from '../../../../../common/hooks/useNavigateTo';
+
+export const DisclaimerDialog = () => {
+  const [disclaimerAgreed, setDisclaimerAgreed] = useState(false);
+  const strings = useStrings();
+  const navigate = useNavigateTo();
+  const { openModal, closeModal } = useModal();
+  const localStorage = new LocalStorageApi();
+
+  const onAcceptDisclaimer = () => {
+    localStorage.setSwapDisclaimerModalClosed('true');
+    closeModal();
+  };
+  const action = {
+    onClick: onAcceptDisclaimer,
+    disabled: !disclaimerAgreed,
+    primary: true,
+    label: strings.disclaimerProceed,
+  };
+
+  useEffect(() => {
+    const checkModalState = async () => {
+      try {
+        const wasClosed = await localStorage.getSwapDisclaimerModalClosed();
+        if (wasClosed === undefined || wasClosed === 'false') {
+          openModal({
+            title: strings.disclaimerTitle,
+            content: (
+              <DisclaimerDialogBody
+                disclaimerAgreed={disclaimerAgreed}
+                setDisclaimerAgreed={setDisclaimerAgreed}
+                action={action}
+              />
+            ),
+            height: '588px',
+            width: '702px',
+            modalId: 'swapDisclaimer',
+            onClose: () => {
+              navigate.walletTransactions();
+              localStorage.setSwapDisclaimerModalClosed('false');
```

Comment:
make the above func param a boolean then just set false, setting 'false' string isn't good practice
```suggestion
              localStorage.setSwapDisclaimerModalClosed(false);
```